### PR TITLE
Improve requirejs load files

### PR DESCRIPTION
**Pull request by mrtuvn**
### Description (*)

Instead always load by default we require it inside function scope by require. For some components such as (confirm, alert..)

Precondition
Production
all cache enabled
Work same old code. Reduce requests js load

### Fixed Issues (if relevant)

1. Fixes magento/magento2#<issue_number>

### Manual testing scenarios (*)

1. ...
2. ...

### Questions or comments

CC: @krzksz 
Any feedback would be apperciated! Thanks

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by ihor-sviziev** [comment]
FYI it's partly duplicates https://github.com/magento/magento2/pull/30647

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run Static Tests, Integration Tests, Functional Tests CE, Functional Tests EE

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run Functional Tests B2B

**Comment by Den4ik** [comment]
@magento run Unit Tests, Functional Tests EE, Functional Tests B2B

**Comment by Den4ik** [comment]
@mrtuvn Could you please check failed unit tests

**Review by eduard13 — CHANGES_REQUESTED** [review]
Hi @mrtuvn, can you please separate the RequireJs upgrade as a separate PR and do all the rest of improvements separately later?

Thank you.

**Comment by mrtuvn** [comment]
Sure! Will revert that

**Comment by eduard13** [comment]
@mrtuvn, looks like it got a bit messed up right now. Could you please open a new PR with RequireJS upgrade only?
Thank you.

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by Den4ik** [comment]
@mrtuvn I'm not sure that we should include PRs to 2.4-develop branch into "Related Pull Requests" section because as I'm correct they are merging for running the tests and looks like it's the reason for failed static tests.
@sivaschenko Is I correct?

**Comment by Den4ik** [comment]
https://github.com/magento/magento2/pull/33541

**Comment by eduard13** [comment]
@Den4ik as far as I know, relating PRs doesn't work within the same repository. We can link CE - EE, but not CE - CE. So that section doesn't have any effect in this case. However, I might be wrong.

@mrtuvn could you please open a **new** PR with RequireJS upgrade only ⚠️ ?

**Comment by mrtuvn** [comment]
@magento run Static Tests, Functional Tests B2B, Functional Tests CE

**Comment by mrtuvn** [comment]
@Den4ik related PR functional seem not working as expected. But thank for let me know that
@eduard13 i have created in another seperate PR

**Comment by Den4ik** [comment]
@magento run Functional Tests B2B

**Comment by ihor-sviziev** [comment]
@magento run Functional Tests B2B

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run Functional Tests CE, Static Tests

**Comment by mrtuvn** [comment]
Squash commits
@magento run all tests

**Comment by mrtuvn** [comment]
Squash commits + rebase branch
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run Functional Tests EE

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run WebAPI Tests, Functional Tests EE, Functional Tests CE

**Comment by ihor-sviziev** [comment]
@magento run Functional Tests CE

**Review by ihor-sviziev — APPROVED** [review]

**Comment by ihor-sviziev** [comment]
@sidolov, this PR reduces amount of loaded files and improving frontend performance. I think priority should be increased to P2

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-9184](https://jira.corp.magento.com/browse/ENGCOM-9184) has been created to process this Pull Request
:eight_spoked_asterisk: @ihor-sviziev, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by mrtuvn** [comment]
Just find one file 
https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/PageCache/view/frontend/web/js/page-cache.js#L10-L72

component `consoleLogger` always requested by default

**Comment by ihor-sviziev** [comment]
@mrtuvn TBH I think we can remove it after https://github.com/magento/magento2/pull/32244

**Review by Den4ik — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @Den4ik, thank you for the review.
[ENGCOM-9184](https://jira.corp.magento.com/browse/ENGCOM-9184) has been created to process this Pull Request

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run Functional Tests CE, Functional Tests B2B, Functional Tests EE

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run Unit Tests

File: app/code/Magento/PageCache/view/frontend/web/js/page-cache.js [hunk 1]
```diff
@@ -45,31 +44,31 @@ define([
...
+                    // prevent cross origin iframe content reading
+                    if ($(element).prop('tagName') === 'IFRAME') {
+                        iframeHostName = $('<a>').prop('href', $(element).prop('src'))
+                            .prop('hostname');
+
+                        if (window.location.hostname !== iframeHostName) {
+                            return [];
+                        }
+                    }
+
```
Comment (ihor-sviziev) [inline app/code/Magento/PageCache/view/frontend/web/js/page-cache.js]: looks like it should be for `el`
```suggestion
                    var iframeHostName;

                    // prevent cross origin iframe content reading
                    if ($(el).prop('tagName') === 'IFRAME') {
                        iframeHostName = $('<a>').prop('href', $(el).prop('src'))
                            .prop('hostname');

                        if (window.location.hostname !== iframeHostName) {
                            return [];
                        }
                    }

```

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

File: app/code/Magento/PageCache/view/frontend/web/js/page-cache.js (hunk 1 above)
Comment (mrtuvn) [reply app/code/Magento/PageCache/view/frontend/web/js/page-cache.js]: Updated as your suggestion

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run Integration Tests, Unit Tests

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run Static Tests, Unit Tests

**Comment by engcom-Hotel** [comment]
Hello @mrtuvn,

I can see the PR is in the Draft status. Can you please let us know if is there anything pending in this PR?

Thanks

**Comment by mrtuvn** [comment]
Sorry but current i don't have enought spare time for continue with this. Feel free for all to create new PR for this if interested

**Comment by engcom-Hotel** [comment]
> Sorry but current i don't have enough spare time for continue with this. Feel free for all to create new PR for this if interested

**Comment by engcom-Hotel** [comment]
Hello @mrtuvn,

As per your comment [here](https://github.com/magento/magento2/pull/33483#issuecomment-1572203286), we are closing this PR:

> Sorry but current i don't have enough spare time for continue with this. Feel free for all to create new PR for this if interested

Whenever you want to resume on this PR, you can reopen it or ask us to reopen.

Thanks

**Pull request closed without merging**